don't choke on utf8 content

amireh opened the pull request:
css files that contain utf-8 characters like ▲ were tripping up the
extractor, but only in specific environments

test plan: the specs pass in an environment like the following:

```bash
docker pull dkamer/canvas-lms-base@sha256:d55d85b8d84a9c1e0d57d865e49c562dcfb0b6be4eae10946722436386330ecb
docker run --rm -it \
  -v "$PWD":/usr/src/app:ro \
  -v "$PWD"/tmp:/usr/src/app/tmp:rw \
  -w /usr/src/app \
  dkamer/canvas-lms-base@sha256:d55d85b8d84a9c1e0d57d865e49c562dcfb0b6be4eae10946722436386330ecb \
    sh -c 'bundle install && bundle exec rspec spec/extractor_spec.rb'
```

amireh committed:
don't choke on utf8 content

css files that contain utf-8 characters like ▲ were tripping up the
extractor, but only in specific environments

test plan: the specs pass in an environment like the following:

    docker pull dkamer/canvas-lms-base@sha256:d55d85b8d84a9c1e0d57d865e49c562dcfb0b6be4eae10946722436386330ecb
    docker run --rm -it \
      -v "$PWD":/usr/src/app:ro \
      -v "$PWD"/tmp:/usr/src/app/tmp:rw \
      -w /usr/src/app \
      dkamer/canvas-lms-base@sha256:d55d85b8d84a9c1e0d57d865e49c562dcfb0b6be4eae10946722436386330ecb \
        sh -c 'bundle install && bundle exec rspec spec/extractor_spec.rb'

amireh commented:
refs https://github.com/instructure/canvas-lms/issues/1836

amireh committed:
apply fix also to generator

although it's not as important as for the extractor, since the generator
only consumes the template, it won't hurt in case the template ends up
having utf8 content too

amireh committed:
bump to 1.2.1

mziwisky commented on spec/spec_helper.rb:
```diff
@@ -32,3 +32,27 @@ def create_test_file
 end
 
+
+def create_utf8_test_file
```
i know this is just copying the method established above, but maybe we could take this opportunity to clean these up a bit with Tempfile https://ruby-doc.org/stdlib-2.5.3/libdoc/tempfile/rdoc/Tempfile.html

`Tempfile.open` cleans up after itself, and lets us use the system's temp dir so we don't have to create a local `tmp` (could delete that from `.gitignore` after this, i think), and even takes an `encoding` option so we can be explicit about making it utf-8.  (i know that's the default anyway, but it's nice to anchor that in code)

this expands the scope of this commit a bit, and it ain't broke, so i'm not hardline on it.

amireh replied:
will do

mziwisky reviewed (approved):
non-blocking suggestion

amireh committed:
inline the test fixtures for extractor spec

this cleans up the spec helpers to use tempfiles instead, consequently
you don't need to mount ./tmp anymore in the command earlier in the
series:

    docker run --rm -it \
      -v "$PWD":/usr/src/app:ro \
      -w /usr/src/app \
      dkamer/canvas-lms-base@sha256:d55d85b8d84a9c1e0d57d865e49c562dcfb0b6be4eae10946722436386330ecb \
        sh -c 'bundle install && bundle exec rspec spec/extractor_spec.rb'

amireh committed:
stop git-ignoring tmp/
